src/garage/_environment.py

```diff
@@ -74,6 +74,20 @@ def max_episode_length(self):
         """
         return self._max_episode_length
 
+    def __eq__(self, other):
+        """See object.__eq__.
+
+        Args:
+            other (EnvSpec): the EnvSpec to compare with.
+
+        Returns:
+            True: if the attributes of EnvSpec are the same.
+        """
+        return ((self.observation_space, self.action_space,
+                 self.max_episode_length) == (other.observation_space,
+                                              other.action_space,
+                                              other.max_episode_length))
```

Comment:
```suggestion
        return (
            self.observation_space == other.observation_space and
            self.action_space == other.action_space and
            self.max_episode_length == other.max_episode_length
        )
```

I'm surprised this isn't handled by namedtuple

Reply:
i think the default `__eq__` checks if they are the exact same instance (via looking at the address?). `==` wouldn't return true for `EnvSpec` with same values without this magic function change.